File: study-restful/src/main/java/com/example/demo/controller/HelloWorldController.java
package com.example.demo.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.demo.bean.HelloWorldBean;

@RestController
@RequestMapping("/")
public class HelloWorldController {

	@GetMapping("/hello-world")
	public String helloWorld() {
		return "hello world!!!!!!";
	}

	@GetMapping("/hello-world-bean")
	public HelloWorldBean helloWorldBean() {
		return new HelloWorldBean("hello world - changed changed");
	}

	@GetMapping("/hello-world-bean/path-variable/{name}")
	public HelloWorldBean helloWorldPathVariable(@PathVariable String name) {
		return new HelloWorldBean(String.format("hello world!!!!!, %s", name));
	}


}

File: src/main/kotlin/com/example/studyrestful/HelloWorldController.kt
package com.example.studyrestful

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class HelloWorldController {

    @GetMapping("/hello-world")
    fun helloWorld() :String {
        return "hello world!!!!!!sffsfs"
    }

}